orbit/diagnostics/plot.py

```diff
@@ -536,3 +310,133 @@ def plot_bt_predictions(bt_pred_df, metrics=smape, split_key_list=None,
+                os.path.abspath(fig_dir)
+            ))
+        fig_paths = list()
+
+    metric_vals = bt_pred_df.groupby('split_key').apply(lambda x:
+                                                        metrics(x[~x['training_data']][BacktestFitKeys.ACTUAL.value],
+                                                                x[~x['training_data']]['prediction']))
+
+    if split_key_list is None:
+        split_key_list_ = bt_pred_df['split_key'].unique()
+    else:
+        split_key_list_ = split_key_list
+
+    if fix_xylim:
+        all_values = np.concatenate((bt_pred_df['actual'].values, bt_pred_df['prediction'].values))
+        ylim = (np.min(all_values) * 0.99, np.max(all_values) * 1.01)
+        xlim = (bt_pred_df['date'].values[0], bt_pred_df['date'].values[-1])
+
+    for idx, split_key in enumerate(split_key_list_):
+        fig, ax = plt.subplots(1, 1, figsize=figsize)
+        tmp = bt_pred_df[bt_pred_df['split_key'] == split_key].copy()
+        ax.plot(tmp['date'], tmp['prediction'],
+                color=PredPal.PREDICTION_LINE.value, lw=lw)
+
+        train_df = tmp.loc[tmp['training_data'], :]
+        ax.scatter(train_df['date'],
+                   train_df[BacktestFitKeys.ACTUAL.value],
+                   marker='.', color=PredPal.ACTUAL_OBS.value, alpha=0.8, s=markersize,
+                   label='train response')
+
+        test_df = tmp.loc[~tmp['training_data'], :]
+        ax.scatter(test_df['date'],
+                   test_df[BacktestFitKeys.ACTUAL.value],
+                   marker='.', color=PredPal.TEST_OBS.value, alpha=0.8, s=markersize,
+                   label='test response')
+
+        ax.set_title(label='split {}; {} {:.3f}'. \
+                     format(split_key, metrics.__name__, metric_vals[split_key]))
+        if include_vline:
+            cutoff_date = tmp[~tmp['training_data']]['date'].min()
+            ax.axvline(x=cutoff_date, linestyle='--',
+                       color=PredPal.HOLDOUT_VERTICAL_LINE.value,
+                       alpha=.8)
+        if fix_xylim:
+            ax.set_xlim(xlim)
+            ax.set_ylim(ylim)
+
+        ax.legend()
+        plt.suptitle(title, fontsize=fontsize)
+        fig.tight_layout()
+
+        if fig_dir:
+            fig_path = '{}/splits_{}.png'.format(fig_dir, idx)
+            fig_paths.append(fig_path)
+            fig.savefig(fig_path)
+        if is_visible:
+            plt.show()
+        else:
+            plt.close()
+
+    if fig_paths and export_gif:
+        import imageio
+        with imageio.get_writer('{}/orbit-backtest.gif'.format(fig_dir), mode='I', **imageio_args) as writer:
+            for fig_path in fig_paths:
+                image = imageio.imread(fig_path)
+                writer.append_data(image)
+
+# TODO: update palatte
+@orbit_style_decorator
+def metric_horizon_barplot(df, model_col='model', pred_horizon_col='pred_horizon',
```

Comment:
Enum for 'model', 'pred_horizon' and etc

Reply:
i am not sure if we should do that for function signature as it goes into function hint and docstring etc.